modules/multi2vec-bind/vectorizer/vectorizer.go

```diff
@@ -131,6 +131,11 @@ func (v *Vectorizer) object(ctx context.Context, id strfmt.UUID,
 	// vectorize image and text
 	var texts, images, audio, video, imu, thermal, depth []string
 	if schema != nil {
+		appendTexts := func(text, prop string) {
+			texts = append(texts, text)
+			vectorize = vectorize || (objDiff != nil && objDiff.IsChangedProp(prop))
+		}
```

Comment:
Are the changes in this file, and in `modules/multi2vec-clip/vectorizer/vectorizer.go` part of what will be solved with #3950? Or are these module changes unrelated to that issue?

Reply:
Adjustments were required due to changes made in `properties_validation.go`:  `stringArrayVal(...)` now returns `[]string` instead of `[]interface{}`. While remaining vectorizers accepted both `[]string` and `[]interface{}`, those 2 handled only `[]interface{}`.

While I took the opportunity to refactor code a bit and extract common part to func, the change was about accepting `[]string` together with `[]interface` and `string`.

Reply:
`multi2vec-clip` was also missing check if array prop is in fact selected for vectorization (was outside of `if ichek.TextField(prop)` condition)